ArcGISRuntimeSDKQt_CppSamples/Maps/MapReferenceScale/MapReferenceScale.cpp

```diff
@@ -86,7 +82,9 @@ void MapReferenceScale::setCurrentMapScale(double scale)
 void MapReferenceScale::setMapScaleToReferenceScale(double scale)
 {
   if(m_mapView)
-    m_mapView->setViewpointScale(scale);
+    m_mapView->setViewpointScaleAsync(scale).then(this, [this](bool){
+      emit currentMapScaleChanged();
+    });
```

Comment:
You don't need the .then on this. You should keep the `mapScaleChanged` handler above to react the changes.